test/check_sharding_data_test.lua

```diff
@@ -88,6 +88,55 @@ function g.test_sharding_key_ok()
 end
 
 
+function g.test_sharding_func_ok()
+    local space = table.deepcopy(test_space)
+    space.indexes = {primary_index}
+    local schema = {
+        spaces = {
+            space = space
+        }
+    }
+
+    -- strcrc32
+    g.space.sharding_func = 'strcrc32'
+    local res, err = ddl.check_schema(schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(res, true)
+
+    -- vshard.router.bucket_id_strcrc32()
+    g.space.sharding_func = 'vshard.router.bucket_id_strcrc32()'
+    local res, err = ddl.check_schema(schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(res, true)
+
+    -- mpcrc32
+    g.space.sharding_func = 'mpcrc32'
+    local res, err = ddl.check_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(res, true)
+
+    -- vshard.router.bucket_id_mpcrc32()
+    g.space.sharding_func = 'vshard.router.bucket_id_mpcrc32()'
```

Comment:
Please, use just name, without parentheses.

Reply:
Updated:

In `ddl/check.lua`
```lua
local function check_sharding_func(space)
    if not space.sharding_func then
        return true
    end

    do -- check that sharding_func format is valid
        ...
        if type(space.sharding_func) == 'string' then
            local known_sharding_funcs = {
                ['vshard.router.bucket_id_strcrc32'] = true,
                ['vshard.router.bucket_id_mpcrc32'] = true,
            }
        ...
        end
    ...
    end
...
end
```

And in `test/`

```lua
function g.test_vshard_sharding_func()
    -- vshard.router.bucket_id_strcrc32
    g.space.sharding_func = 'vshard.router.bucket_id_strcrc32'
    local res, err = ddl.check_schema(g.schema)
    t.assert_equals(err, nil)
    t.assert_equals(res, true)

    -- vshard.router.bucket_id_mpcrc32
    g.space.sharding_func = 'vshard.router.bucket_id_mpcrc32'
    local res, err = ddl.check_schema(g.schema)
    t.assert_equals(err, nil)
    t.assert_equals(res, true)
end
```